### PR TITLE
Rzhou/us124158 rubric criterion action menu

### PR DESCRIPTION
**Pull request by rebeccazhou0727**
The Action menu should only appear when the `rubrics-criterion-action` feature flag is turned on while the remove trash-can icon will be hidden. We would need to reformat d2l-rubric-criterion-editor (specifically `gutter-right`) once we decide to remove this functionality permanently.

Expected Behavior: 

- Keyboard focus should be directly given to the input text area of the copied criterion
- Action menu being hidden when the  `rubrics-criterion-action`  is turned off or the criterion cannot be copied nor deleted

Added lang terms

- actionsforCriterion
- copiedCriterionLoading
- copyCriterion
- copyRow
- criterionCopied
- deleteCriterion
- deleteRow

Rally Ticket:
https://rally1.rallydev.com/#/57732444928d/custom/373260458992?detail=%2Fuserstory%2F484021014724

### CODE REVIEW

File: editor/d2l-rubric-criterion-editor.js [hunk 1]
```diff
@@ -505,15 +527,26 @@ Polymer({
 		},
 		_hierarchicalHeight: {
 			type: Number
+		},
+		rubricsCriterionAction: {
```
Comment (MitchelPaulin) [inline editor/d2l-rubric-criterion-editor.js:531]: What does this variable represent?

File: editor/d2l-rubric-criterion-editor.js [hunk 2]
```diff
@@ -786,6 +823,40 @@ Polymer({
...
+			this.dispatchEvent(new CustomEvent('d2l-rubric-criterion-copied', {
+				bubbles: true,
+				composed: true,
+			}));
+			setTimeout(function() {
+				announce('criterionCopied');
+			}.bind(this), 2000);
+
+			this.perfMark(`criterionAddedEnd=${uuid}`);
+			this.logCriterionCopiedAction(`criterionCopiedSatrted-${uuid}`, `criterionCopiedEnd=${uuid}`);
```
Comment (Callum-Mitchell) [inline editor/d2l-rubric-criterion-editor.js:845]: `criterionCopiedSatrted-${uuid}` - typo?

**Review by Callum-Mitchell — COMMENTED** [review]
Nice work! Just one thing to check - see comment.

File: editor/d2l-rubric-criterion-editor.js [hunk 3]
```diff
@@ -797,6 +834,40 @@ Polymer({
...
+
+		const uuid = this.getUUID();
+		this.perfMark(`criterionCopiedStarted-${uuid}`);
+		announce(this.localize('copiedCriterionLoading', 'name', this._criterionName));
+		this.performSirenAction(action).then(function() {
+			this.dispatchEvent(new CustomEvent('d2l-rubric-criterion-copied', {
+				bubbles: true,
+				composed: true,
+			}));
+			setTimeout(function() {
```
Comment (rgoessl) [inline editor/d2l-rubric-criterion-editor.js:851]: What is this timeout for? Does the UI take a while to render after this succeeds?

File: editor/d2l-rubric-criterion-editor.js [hunk 4]
```diff
@@ -797,6 +834,40 @@ Polymer({
...
+		var action = this.entity.getActionByName('copy');
+		if (!action) return;
+		var copyButton = e.currentTarget;
+		copyButton.setAttribute('disabled', '');
+
+		const uuid = this.getUUID();
+		this.perfMark(`criterionCopiedStarted-${uuid}`);
+		announce(this.localize('copiedCriterionLoading', 'name', this._criterionName));
+		this.performSirenAction(action).then(function() {
+			this.dispatchEvent(new CustomEvent('d2l-rubric-criterion-copied', {
```
Comment (rgoessl) [inline editor/d2l-rubric-criterion-editor.js:847]: Is this event used?

File: editor/d2l-rubric-criteria-groups-editor.js [hunk 5]
```diff
@@ -130,6 +131,7 @@ Polymer({
 			observer: '_totalScoreChanged'
 		},
 
+		rubricsCriterionAction: Boolean,
```
Comment (rgoessl) [inline editor/d2l-rubric-criteria-groups-editor.js:134]: I'm assuming this defaults to `value: false`?

**Review by rgoessl — CHANGES_REQUESTED** [review]

File: editor/d2l-rubric-criterion-editor.js (hunk 3 above)
Comment (rebeccazhou0727) [reply editor/d2l-rubric-criterion-editor.js:851]: The `setTimeout `was in the `_handleAddCriterion `function so I thought the behaviour may be similar for copy

File: editor/d2l-rubric-criterion-editor.js (hunk 4 above)
Comment (rebeccazhou0727) [reply editor/d2l-rubric-criterion-editor.js:847]: The event is not used anymore but similar events are created for d2l-rubric-criterion-deleted and d2l-rubric-criterion-added and they are not used anywhere in this repository. Should I just delete them?

Comment (rgoessl) [reply editor/d2l-rubric-criterion-editor.js:847]: It doesn't look like any of them are used. Should be safe to delete

File: editor/d2l-rubric-criterion-editor.js (hunk 3 above)
Comment (rgoessl) [reply editor/d2l-rubric-criterion-editor.js:847]: It's probably fine then

**Review by rgoessl — APPROVED** [review]

**Pull request merged**